OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,391 @@
+				FadeInCountParticle(wr);
+			else if (particleCountFader == ParticleCountFaderType.FadeOut)
+				FadeOutCountDoPa(wr);
+		}
+
+		void FadeInCountParticle(WorldRenderer wr)
+		{
+			SpawnParticles(1, 0, 100);
+
+			// Remove Particles, which are getting replaced from the top to the bottom by the "EdgeCheckReplace",
+			// when scrolling down, as long as the FadeIn is not completed,
+			// to avoid having particles at the top and bottom, but not in the middle of the screen.
+			for (int i = 0; i < ParticleList.Count; i++)
+			{
+				if (ParticleList[i].PosY < 0)
+					ParticleList.RemoveAt(i);
+			}
+
+			// Add Particles when the weather is fading in and scrolling up, to fill areas above
+			if (antiScrollPrevTopLeft.Y > wr.Viewport.TopLeft.Y)
+			{
+				// Get delta Y and limit to the max value
+				var tempRangeY = antiScrollPrevTopLeft.Y - wr.Viewport.TopLeft.Y;
+				var tempParticleCount = CalculateParticleCount(Game.Renderer.Resolution.Width, (int)tempRangeY);
+				if (ParticleList.Count + tempParticleCount > maxParticleCount)
+					tempParticleCount = maxParticleCount - ParticleList.Count;
+				SpawnParticles(tempParticleCount, (int)tempRangeY, 50);
+			}
+		}
+
+		void FadeOutCountDoPa(WorldRenderer wr)
+		{
+			for (int i = 0; i < ParticleList.Count; i++)
+			{
+				if (ParticleList[i].PosY > (Game.Renderer.Resolution.Height - ParticleList[i].Gravity))
+					ParticleList.RemoveAt(i);
+			}
+		}
+
+		Particle XAxisSwing(Particle tempParticle)
+		{
+			// Left direction turn
+			if (tempParticle.SwingOffset < (-1 * tempParticle.SwingAmplitude))
+				tempParticle.SwingDirection = !tempParticle.SwingDirection;
+
+			// Right direction turn
+			else if (tempParticle.SwingOffset > tempParticle.SwingAmplitude)
+				tempParticle.SwingDirection = !tempParticle.SwingDirection;
+
+			// Performe the X-Axis-Swing
+			if (tempParticle.SwingDirection)
+				tempParticle.SwingOffset += tempParticle.SwingSpeed;
+			else
+				tempParticle.SwingOffset -= tempParticle.SwingSpeed;
+
+			return tempParticle;
+		}
+
+		public void Tick(Actor self)
+		{
+			windTickCountdown--;
+		}
+
+		Particle WindLogic(Particle tempParticle)
+		{
+			if (info.ChangingWind)
+			{
+				if (windTickCountdown <= 0)
+				{
+					windTickCountdown = Game.CosmeticRandom.Next(info.WindTick[0], info.WindTick[1]);
+					if (Game.CosmeticRandom.Next(0, 2) == 1)
+					{
+						if (currentWindIndex > 0)
+						{
+							currentWindIndex--;
+							targetWindXOffset = info.WindStrength[currentWindIndex];
+						}
+					}
+					else
+					{
+						if (currentWindIndex < info.WindStrength.Length - 1)
+						{
+							currentWindIndex++;
+							targetWindXOffset = info.WindStrength[currentWindIndex];
+						}
+					}
+				}
+			}
+			else
+				targetWindXOffset = info.WindStrength[0];
+
+			// Fading the wind in little steps towards the TargetWindOffset
+			if (!info.InstantWindChanges)
+			{
+				if (currentWindXOffset != targetWindXOffset)
```

Comment:
same here, change !info to info, switch the if and else body and change the else to else if (currentWindXOffset != targetWindXOffset).